<p>I am trying to connect from SERVER-1 to my mongo server which is running in some other sever SERVER-2 So my grails app is running in SERVER-1. So I have given external configuration like this.</p>  <pre><code>grails {     mongo {         host = "&lt;SERVER-2&gt;-host"         port = 27017         username = "myapp"         password = "myapp"         databaseName = "myapp"         options {             autoConnectRetry = true             connectTimeout = 3000         }     } } </code></pre>  <p>And in the SERVER-2 I've created myapp db and with the same user and credentials</p>  <pre><code>use myapp db.createUser( { "user" : "myapp",                  "pwd": "myapp",                  "roles" : [] },                { w: "majority" , wtimeout: 5000 } ) </code></pre>  <p>And am able to see the users list like below</p>  <pre><code>&gt; db.getUsers() [     {         "_id" : "myapp.myapp",         "user" : "myapp",         "db" : "myapp",         "roles" : [ ]     } ] </code></pre>  <p>Mongo configuration contains "noauth=true", And from SERVER-1, I'm able to connect to SERVER-2 mongo using the below command </p>  <pre><code>mongo SERVER-2-HOST:27017/myapp  -u myapp -p myapp </code></pre>  <p>But when i try to connect from SERVER-1 grails application, its giving the below error</p>  <pre><code>| Error Error executing script LoadVars: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'transactionManagerPostProcessor': Initialization of bean failed; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'transactionManager': Cannot resolve reference to bean 'mongoTransactionManager' while setting constructor argument with key [1]; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'mongoTransactionManager': Cannot resolve reference to bean 'mongoDatastore' while setting bean property 'datastore'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'mongoDatastore': FactoryBean threw exception on object creation; nested exception is org.springframework.data.mongodb.CannotGetMongoDbConnectionException: Failed to authenticate to database [myapp], username = [myapp], password = [m***p] </code></pre>  <p>My mongod.conf file</p>  <pre><code># mongod.conf  logpath=/var/log/mongodb/mongod.log  logappend=true  fork=true  #port=27017  dbpath=/var/lib/mongo  pidfilepath=/var/run/mongodb/mongod.pid  # Listen to local interface only. Comment out to listen on all interfaces. #bind_ip=127.0.0.1  # nojournal=true  #cpu=true  #noauth=false auth=true </code></pre>  <p>Am I missing anything here?</p>